samza-core/src/main/java/org/apache/samza/job/model/JobModel.java

```diff
@@ -39,7 +38,6 @@
  * an id, partition information, etc.
  * </p>
  */
-@JsonIgnoreProperties(ignoreUnknown = true)
 public class JobModel {
```

Comment:
Yep, this line was moved to the mixin, which is used to add annotations to JobModel. So it should be the same behavior. This change decouples JobModel from Jackson.
There is a unit test for deserializing with unknown fields, so that verifies the functionality.